db/db_iter.cc

```diff
@@ -217,6 +217,9 @@ class DBIter final: public Iterator {
         *prop = "Iterator is not valid.";
       }
       return Status::OK();
+    } else if (prop_name == "rocksdb.iterator.internal-key") {
+      *prop = saved_key_.GetUserKey().ToString();
```

Comment:
Should be `ToString(true)`.

Reply:
Hmm, I guess `false` may also work.

Reply:
Yes, I think HEX encoding is not necessary, so we should be fine with default, which is false.

Reply:
I don't think it is not necessary to be in hex, as we don't necessarily need it to be printed. string itself can contain un-printable characters (including null). I verified that we are indeed using the right string api based on the *size* of `data`, instead of just a regular copy (which stops the first time you encounter a null character): 
https://github.com/facebook/rocksdb/blob/master/util/slice.cc#L158 . Do you agree?
I tested it earlier that we are indeed returning the full slice value converted to a string.